third_party/subscriptions-project/config.js

```diff
@@ -955,7 +875,7 @@ class JsonLdParser {
       possibleConfigs = [possibleConfigs];
     }
 
-    for (let i = 0; i < possibleConfigs.length; i++) {
+    for (let i = 0; i < /** @type {Array} */ (possibleConfigs).length; i++) {
```

Comment:
nitpicks (feel free to ignore):
1) What is this an array of `/** @type {Array<some type?>}*/`?
2) Making this a for in loop would remove the assignment required on like 879 and prevent `i` being overloaded be the inner loop on line 890.

Reply:
I'll follow up in another PR!

Reply:
1) https://github.com/subscriptions-project/swg-js/pull/1564 updates the type to `!Array{!JsonObject}`

2) Would a for-in loop remove the assignment? I know a for-of loop would, but currently we can only use those in tests and build-system stuff

Reply:
Forgot about that, I pretty much just work in build-system. A `forEach` loop would also work.

Reply:
`find()` might be better, since we want to terminate the loop after finding a match

`forEach()` loops only terminate on exceptions

Reply:
TBH I hadn't really tried to understand what the code was doing. A find would be great. Maybe something like this?
```
const {config, productId} = possibleConfigs
.filter(config => this.checkType_.checkValue(config['@type'], ALLOWED_TYPES))
.map(config => {
  const partOfArray = this.valueArray_(config, 'isPartOf');
  if (!partOfArray) {
    return;
  }
  for (let i = 0; i < partOfArray.length; i++) {
    productId = this.discoverProductId_(partOfArray[i]);
    if (productId) {
      return {config, productId};
    }
  }
})
.find(Boolean) || {};

const isAccessibleForFree = this.bool_(
  this.singleValue_(config, 'isAccessibleForFree'),
  /* default */ true
);

return new PageConfig(productId, !isAccessibleForFree);
```